File: api/internal/utils/id_card/aliyun_id_card.go
package id_card

import (
	"context"
	"errors"

	"github.com/gogf/gf/v2/encoding/gjson"
	"github.com/gogf/gf/v2/frame/g"
	"github.com/gogf/gf/v2/net/gclient"
	"github.com/gogf/gf/v2/util/gconv"
)

type AliyunIdCard struct {
	Ctx     context.Context
	Host    string `json:"aliyunIdCardHost"`
	Path    string `json:"aliyunIdCardPath"`
	Appcode string `json:"aliyunIdCardAppcode"`
}

func NewAliyunIdCard(ctx context.Context, config map[string]interface{}) *AliyunIdCard {
	aliyunIdCardObj := AliyunIdCard{
		Ctx: ctx,
	}
	gconv.Struct(config, &aliyunIdCardObj)
	return &aliyunIdCardObj
}

func (idCardThis *AliyunIdCard) Auth(idCardName string, idCardNo string) (idCardInfo map[string]interface{}, err error) {
	// res, err := idCardThis.CreateClient().Get(idCardThis.Ctx, idCardThis.Host+idCardThis.Path+`?cardno=`+idCardNo+`&name=`+url.QueryEscape(idCardName))
	res, err := idCardThis.CreateClient().Get(idCardThis.Ctx, idCardThis.Host+idCardThis.Path, g.Map{
		`cardno`: idCardNo,
		`name`:   idCardName,
	})
	if err != nil {
		return
	}
	defer res.Close()
	resData := gjson.New(res.ReadAllString())
	if resData.Get(`resp.code`).Int() != 0 {
		err = errors.New(resData.Get(`resp.desc`).String())
		return
	}
	idCardInfo = resData.Get(`data`).Map()
	return
}

func (idCardThis *AliyunIdCard) CreateClient() (client *gclient.Client) {
	client = g.Client().SetHeaderMap(g.MapStrStr{`Authorization`: `APPCODE ` + idCardThis.Appcode})
	return
}

File: api_sw/app/Plugin/IdCard/AliyunIdCard.php
<?php

declare(strict_types=1);

namespace App\Plugin\IdCard;


class IdCardOfAliyun extends AbstractIdCard
{
    protected $config = [];

    /**
     * 实名认证
     *
     * @param string $idCardName
     * @param string $idCardNo
     * @return array
     */
    public function auth(string $idCardName, string $idCardNo): array
    {
        $client = getHttpClient(['timeout' => 5]);
        $option = [
            'headers' => ['Authorization' => 'APPCODE ' . $this->config['appcode']],
            'query' => ['name' => $idCardName, 'cardno' => $idCardNo]
            // 'json' => ['key' => 'value'],
            // 'form_params' => ['key' => 'value']
        ];
        try {
            $url = $this->config['host'] . $this->config['path'];
            $response = $client->get($url, $option);
            // $response = $client->post($url, $option);
            $resData = $response->getBody()->getContents();
            $resData = json_decode($resData, true);
        } catch (\Throwable $th) {
            throwFailJson(79999999, $th->getMessage());
        }
        if (!(isset($resData['resp']['code']) && $resData['resp']['code'] == 0)) {
            throwFailJson(79999999, $resData['resp']['desc'] ?? '');
        }

        $idCardInfo = [
            'gender' => 0,
            'address' => $resData['data']['address'],
            'birthday' => $resData['data']['birthday'],
        ];
        switch ($resData['data']['sex']) {
            case '男':
                $idCardInfo['gender'] = 1;
                break;
            case '女':
                $idCardInfo['gender'] = 2;
                break;
        }
        return $idCardInfo;
    }
}
